dGame/dComponents/PetComponent.h

```diff
@@ -357,7 +352,9 @@ class PetComponent : public Component
 	PreconditionExpression* m_Preconditions;
 
 	/**
-	 * The rate at which imagination is drained from the user for having the pet out.
+	 * Pet information loaded from the CDClientDatabase
 	 */
-	float imaginationDrainRate;
+	CDPetComponent& m_PetInfo;
```

Comment:
Query on if there should be such a guarantee. Feels unnecessary to deep copy every time you want to keep information from the cdclient elsewhere.

Reply:
the cdclient database values are destructed before the server starts handling shutdown so this will cause undefined behavior keeping a reference to a destructed element.  Keeping a reference is a bad pattern and is what caused save data to be corrupted for end users when the same thing was done for mission data.

Reply:
for references in a map, sure its fine to keep one around, but thats only if you can guarantee the reference will stay alive, and in the case of the current cdclient implementation this is false.  see #1183 for the details on how cdclient tables are destructed.

Reply:
I should make it clear, I am all _for_ storing the data as a reference, but I already know this is not safe currently.  If the data in the cdclient table was indeed safe to reference, I would have no problem doing so.

Reply:
> I should make it clear, I am all _for_ storing the data as a reference, but I already know this is not safe currently. If the data in the cdclient table was indeed safe to reference, I would have no problem doing so.

Is it worth considering providing such a guarantee, then?

Reply:
Yes it is, but we would need to research how to extend the lifetime of the orm containers to rhe binary life instead of until a signal is handled